External/AvalonTools/AvalonTools.cpp

```diff
@@ -215,7 +215,8 @@ namespace AvalonTools {
   }
 
   unsigned int set2DCoords(ROMol &mol,bool clearConfs){
-    struct reaccs_molecule_t *mp=molToReaccs(mol);
+    auto smiles = MolToSmiles(mol);
+    struct reaccs_molecule_t *mp=stringToReaccs(smiles,true);
```

Comment:
The atoms in the smiles string aren't in the same order as mol, don't you need to grab the indices of the smiles output and remap?

Reply:
I think your example doesn't show this problem as the input is already canonical.

Reply:
You can get the ordering by:

```
std::vector<unsigned int> atomOrdering;
mol.getProp(common_properties::_smilesAtomOutputOrder, atomOrdering);
```

Reply:
And then I think the coords should be:

```
RDGeom::Point3D loc(mp2->atom_array[atomOrdering[i]].x
 ...
```

In any case, we should try some non-canonical smiles in the tests.

Reply:
right, I was just thinking about getting rid of the intermediate ctab, and I completely overlooked the issue with the ordering of atoms. thanks for pointing that out.

Reply:
Just realized that I hadn't commented on this yet.
I'm fine with this once the problem @bp-kelley points out is fixed and a test is added to detect that case (i.e. the input SMILES is in non-canonical order).